src/Xamarin.Android.Build.Tasks/Tests/Xamarin.ProjectTools/Common/TestEnvironment.cs

```diff
@@ -0,0 +1,99 @@
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using Xamarin.Android.Tools.VSWhere;
+
+namespace Xamarin.ProjectTools
+{
+	public static class TestEnvironment
+	{
+		[DllImport ("libc")]
+		static extern int uname (IntPtr buf);
```

Comment:
Instead of P/Invoking **uname**(3), you could just use `Directory.Exists("/Applications")`, as `/Applications` won't exist on non-macOS platforms.

Reply:
I was just moving code which already existed elsewhere so that it could be used outside of test cases. @jonpryor would you prefer that we change this now?